sys/fs/cuse/cuse.c

```diff
@@ -1133,7 +1149,18 @@ cuse_server_ioctl(struct cdev *dev, unsigned long cmd,
 
 		pccmd->entered = curthread;
 
-		*pcmd = pccmd->sub;
+#ifdef COMPAT_FREEBSD64
+		if (!SV_CURPROC_FLAG(SV_CHERI)) {
+			pcmd64 = (struct cuse_command64 *)data;
+			pcmd64->dev = (uint64_t)pccmd->sub.dev;
```

Comment:
When building a hybrid kernel, cc complains about a missing `__cheri_addr` qualifier here.

Reply:
Or just use a `uintptr_t` cast instead of a `uint64_t` cast and then it will DTRT?

Reply:
We have PTROUT for this

Reply:
Hm although that's using uintptr_t not uintcap_t (and you want the latter, as the former will be the same as uint64_t for hybrid kernels...)